Src/GoogleApis.Auth/OAuth2/AuthorizationCodeInstalledApp.cs

```diff
@@ -64,7 +64,7 @@ public ICodeReceiver CodeReceiver
 
             // If the stored token is null or it doesn't have a refresh token and the access token is expired we need 
             // to retrieve a new authorization code.
-            if (token == null || (token.RefreshToken == null && token.IsExpired(flow.Clock)))
+            if (Flow.ShouldForceTokenRetrieval() || token == null || (token.RefreshToken == null && token.IsExpired(flow.Clock)))

```

Comment:
[optional] What do you think? Should we wrap this if in a function
public bool ShouldRequestAuthorizationCode()
{
  return Flow.ShouldForceTokenRetrieval() || token == null || (token.RefreshToken == null && token.IsExpired(flow.Clock));
}

I think it will be neater. If you do so, please add a comment as well.